### DISCUSSION TITLE
Auth flow displays sign in screen while loading

### DISCUSSION THREAD

**Discussion by vcapretz** [expo/router#273] [Q&A]
Hey, I followed the auth example from the documentation and when hooking up a server it displays the sign in screen while the token is being validated in the backend

I tried using `SplashScreen` so it waits before showing anything but that breaks the app.

Here's a video of what I mean: 

https://user-images.githubusercontent.com/14620121/217943240-fb72580e-39db-4c5d-be99-8c27119840a2.mp4

if I use `SplashScreen` in my `app/_layout.tsx` it breaks the app: 
<img width="719" alt="CleanShot 2023-02-09 at 18 24 37@2x" src="https://user-images.githubusercontent.com/14620121/217943504-5ee9f22d-c33d-43b9-bd07-a025e873a08f.png">

here's an example repo with minimum code: https://github.com/vcapretz/expo-router-auth 
essentially I would want something in place while https://github.com/vcapretz/expo-router-auth/blob/ff56ebd91181c35919ca5188ed31722807287dd2/components/auth-provider.tsx#L59 this doesn't resolve

**Comment by TheReidZhang** [comment]
Hey, I am facing the same problem. Have you resolved it? Thanks.
